syntax = "proto3";

import "kernel/dssm_applier/embeddings_transfer/proto/model_embedding.proto";

package NOptimizedModel.NProto;

message TSingleValueStorage {
    string Name = 1;
    string Version = 2;
    float Value = 3;
}

message TNameToVersion {
    string Name = 1;
    string Version = 2;
}

message TSerializedApplyExResult {
    repeated TSingleValueStorage Predicts = 1;
    repeated NEmbeddingsTransfer.NProto.TModelEmbedding Embeddings = 2;
    repeated TSingleValueStorage Parameters = 3;

    repeated TNameToVersion PredictToDefaultVersion = 4;
    repeated TNameToVersion EmbeddingToDefaultVersion = 5;
    repeated TNameToVersion ParameterToDefaultVersion = 6;

    string FlagsVersion = 7;
    repeated TNameToVersion SubmodelsForExport = 8;
}
